### ISSUE TITLE
Energy consumptions problem

### ISSUE THREAD

**Issue by tsedgwick1** [fffonion/tplink-plug-exporter#1]
Hello,

When I want to check metrics datas on my hs110 devices, I use :
`curl -L http://localhost:9233/scrape?target=192.168.0.12`

This is what he shows 

```
curl -L http://localhost:9233/scrape?target=192.168.0.12
# HELP kasa_current Current flowing through device in Ampere.
# TYPE kasa_current gauge
kasa_current{alias="vidéoprojecteur"} 0
# HELP kasa_on_time Time in seconds since online.
# TYPE kasa_on_time counter
kasa_on_time{alias="vidéoprojecteur"} 357610
# HELP kasa_online Device online.
# TYPE kasa_online gauge
kasa_online 1
# HELP kasa_power_load Power of load in Watt.
# TYPE kasa_power_load gauge
kasa_power_load{alias="vidéoprojecteur"} 0
# HELP kasa_power_total Power of load and device itself in Watt.
# TYPE kasa_power_total gauge
kasa_power_total{alias="vidéoprojecteur"} 0
# HELP kasa_relay_state Relay state (switch on/off).
# TYPE kasa_relay_state gauge
kasa_relay_state{alias="vidéoprojecteur"} 1
# HELP kasa_rssi Wifi received signal strength indicator.
# TYPE kasa_rssi gauge
kasa_rssi{alias="vidéoprojecteur"} -35
# HELP kasa_voltage Current voltage connected to device in Volt.
# TYPE kasa_voltage gauge
kasa_voltage{alias="vidéoprojecteur"} 0
```
Do you know why I don't have all the datas ?
My plugs send well all the consumption informations on the android app.

**Comment by dpunkturban** [comment]
Hi,

I had the same problems and had to adjust some variables. 

Can you try this one?

https://github.com/dpunkturban/hs110-prometheus-exporter

If everything works I start a Pull Request

**Comment by aslafy-z** [comment]
Here's a release that contains your fix, for users to test: https://github.com/aslafy-z/tplink-plug-exporter/releases/tag/v0.2.0

Thanks @dpunkturban

**Comment by tsedgwick1** [comment]
Hi,

It works like a charm with this adjustment :)
Thanks @dpunkturban

**Comment by dpunkturban** [comment]
Hi @aslafy-z @tsedgwick1 ,

thanks for your feedback and for building a binary version.

i created the pull request.

br
dennis

**Comment by fffonion** [comment]
Hi all, thanks for the discussion here! It get lost in my inbox somehow:)
Could any of you share with the device type and maybe firmware version you are using?
It seems my US version of HS110 (fw 1.2.5) and HS105 (fw 1.5.4) is still returning metrics without unit, so we might want to support multiple firmwares if it's the case.

**Comment by dpunkturban** [comment]
Hi,

i am using:

- HS 110, Hardware Rev 2.0, Firmware 1.5 (EU Version)
- HS 110, Hardware Rev 2.0, Firmware 1.5.4 (EU Version)

br
dennis

**Comment by fffonion** [comment]
Thank you @dpunkturban ! I guess we can rely on the firmware version to do some conversion then.

**Comment by dpunkturban** [comment]
Sounds good!

Feel free to ask me if you need help with this project.

br
dennis

**Comment by fffonion** [comment]
Hi,
I started a new branch (https://github.com/fffonion/tplink-plug-exporter/tree/emeter) with #2 in and keeps backward compatibilty. Could you test with that branch or with the precompiled binary below?

Thanks
[tplink-plug-exporter_linux_amd64.gz](https://github.com/fffonion/tplink-plug-exporter/files/3923442/tplink-plug-exporter_linux_amd64.gz)

**Comment by fffonion** [comment]
Hi all, the above patch is merged to maste. I just released [v0.2.0](https://github.com/fffonion/tplink-plug-exporter/releases/tag/v0.2.0) that include this and several other fixes. Please feel free to reopen this issue if you are still facing the issue.
Thanks again for catching this!
